java/fury-core/src/main/java/org/apache/fury/type/TypeUtils.java

```diff
@@ -735,9 +742,12 @@ private static LinkedHashSet<Class<?>> listBeansRecursiveInclusive(
         }
       } else if (isCollection(type)) {
         TypeRef<?> elementType = getElementType(typeToken);
+        while (isCollection(elementType.getRawType()) || elementType.getRawType().isArray()) {
```

Comment:
this update is necessary otherwise `List<List<Double>>` tries to inspect `List<Double>` as the element, but really it should be `Double`